tests/baselines/reference/contextualTypingOfTooShortOverloads.types

```diff
@@ -31,11 +31,11 @@ app.use((err: any, req, res, next) => { return; });
 >app.use : IRouterHandler<MyApp> & IRouterMatcher<MyApp>
 >app : MyApp
 >use : IRouterHandler<MyApp> & IRouterMatcher<MyApp>
->(err: any, req, res, next) => { return; } : (err: any, req: any, res: any, next: any) => void
+>(err: any, req, res, next) => { return; } : (err: any, req: Request, res: Response, next: NextFunction) => void

```

Comment:
Previously, this'd get `any` argument types fixed by an earlier overload we didn't actually pick because of an arity mismatch - this is a pretty clear improvement!